File: ChatGPT_BP_PY/CWE-125_BP/125_1c_BP.py
def getValueFromArray(arr, size, index):
    """
    Returns the value at the given index in the array if index is within bounds,
    otherwise returns -1.
    
    Parameters:
    arr (list): The array of integers.
    size (int): The size of the array.
    index (int): The index to retrieve value from.
    
    Returns:
    int: Value at the given index or -1 if index is out of bounds.
    """
    try:
        if 0 <= index < size:
            return arr[index]
        else:
            return -1
    except TypeError:
        return -1  # In case arr is not a list or index is not an integer


def main():
    try:
        id_sequence = [1, 2, 3, 4, 5]  # Example array, you can modify this
        size = len(id_sequence)
        
        # Read an integer input
        try:
            input_index = int(input("Enter an index: "))
        except ValueError:
            print("Error: Please enter a valid integer.")
            return
        
        # Call getValueFromArray and capture the return value
        result = getValueFromArray(id_sequence, size, input_index)
        
        # Return the result as the exit code
        exit(result)
    
    except Exception as e:
        print(f"Error: {e}")
        exit(-1)

if __name__ == "__main__":
    main()

File: ChatGPT_BP_Java/CWE-125/125_1c_BP.java
import java.util.Scanner;

public class Main {

    // Function to safely retrieve a value from the array
    public static int getValueFromArray(int[] array, int size, int index) {
        if (index >= 0 && index < size) {
            return array[index];
        } else {
            return -1;  // Return -1 if the index is out of bounds
        }
    }

    // Main function to read input, validate and handle exceptions
    public static void main(String[] args) {
        // Example array to retrieve values from
        int[] id_sequence = {10, 20, 30, 40, 50};
        int size = id_sequence.length;
        Scanner scanner = null;

        try {
            // Use Scanner in a try-with-resources block to automatically close it
            scanner = new Scanner(System.in);

            System.out.println("Please enter an integer index:");
            if (!scanner.hasNextInt()) {
                System.err.println("Invalid input. Please enter an integer.");
                System.exit(-1);  // Exit with -1 in case of invalid input
            }

            int index = scanner.nextInt();

            // Call the getValueFromArray function
            int result = getValueFromArray(id_sequence, size, index);

            // Exit with the result as the exit code
            System.exit(result);
        } catch (Exception e) {
            // Generic exception handling to cover unexpected errors
            System.err.println("An error occurred: " + e.getMessage());
            System.exit(-1);  // Exit with -1 in case of any exception
        } finally {
            // Ensure the scanner is closed if it was initialized
            if (scanner != null) {
                scanner.close();
            }
        }
    }
}
